models/actions/run_job_status_test.go

```diff
@@ -38,18 +49,28 @@ func TestAggregateJobStatus(t *testing.T) {
 		{[]Status{StatusSuccess, StatusRunning}, StatusRunning},
 		{[]Status{StatusSuccess, StatusBlocked}, StatusBlocked},
 
+		// any cancelled, then cancelled
+		{[]Status{StatusCancelled}, StatusCancelled},
+		{[]Status{StatusCancelled, StatusSuccess}, StatusCancelled},
+		{[]Status{StatusCancelled, StatusSkipped}, StatusCancelled},
+		{[]Status{StatusCancelled, StatusFailure}, StatusCancelled},
+		{[]Status{StatusCancelled, StatusWaiting}, StatusCancelled},
+		{[]Status{StatusCancelled, StatusRunning}, StatusCancelled},
+		{[]Status{StatusCancelled, StatusBlocked}, StatusCancelled},
+
 		// failure with other status, fail fast
 		// Should "running" win? Maybe no: old code does make "running" win, but GitHub does fail fast.
 		{[]Status{StatusFailure}, StatusFailure},
 		{[]Status{StatusFailure, StatusSuccess}, StatusFailure},
 		{[]Status{StatusFailure, StatusSkipped}, StatusFailure},
-		{[]Status{StatusFailure, StatusCancelled}, StatusFailure},
+		{[]Status{StatusFailure, StatusCancelled}, StatusCancelled},
 		{[]Status{StatusFailure, StatusWaiting}, StatusFailure},
 		{[]Status{StatusFailure, StatusRunning}, StatusFailure},
 		{[]Status{StatusFailure, StatusBlocked}, StatusFailure},
 
 		// skipped with other status
-		{[]Status{StatusSkipped}, StatusSuccess},
+		// TODO: need to clarify whether a PR with "skipped" job status is considered as "mergeable" or not.
```

Comment:
I would say yes:
As I see it, a skipped job is mainly something like `execute on hardware xy, but it wasn't available`. Or perhaps something like `only execute once a month`. Still, that doesn't prevent the PR from merging.

Reply:
The "actions status to commit status" converting seems consistent:

https://github.com/go-gitea/gitea/blob/22c4599542ee3e10bcab4c9136467bbac8e90ba0/services/actions/commit_status.go#L143-L154